lightning-invoice/src/lib.rs

```diff
@@ -1429,10 +1424,13 @@ impl Bolt11Invoice {
 
 	/// Returns a list of all fallback addresses as [`Address`]es
 	pub fn fallback_addresses(&self) -> Vec<Address> {
-		self.fallbacks().iter().map(|fallback| {
+		self.fallbacks().iter().filter_map(|fallback| {
 			let payload = match fallback {
 				Fallback::SegWitProgram { version, program } => {
-					Payload::WitnessProgram { version: *version, program: program.to_vec() }
+					match WitnessProgram::new(*version, program.clone()) {
+						Ok(witness_program) => Payload::WitnessProgram(witness_program),
+						Err(_) => return None,
+					}
```

Comment:
So previously you'd accept invalid addresses but now you silently ignore them. Is this even correct? Maybe the error handling should be at creation of `Fallback::SegWitProgram`, so the variant should change to store `WitnessProgram` directly?

Reply:
Perhaps add an API to check if any are invalid so that an application can emit a warning in such case?